rfc-161-technical-direction-in-publishing-2023.md

```diff
@@ -0,0 +1,144 @@
+
+The decisions you make in producing your code should be included in your commit message. It is common that GOV.UK developers look at old commits when trying to resolve problems or understand design decisions. Always include the “why” of a change, the “what” is usually covered by the code.
+
+#### If a migration is worth starting, it’s worth ending
+
+Like many tech organisations, we’ve been burnt by large projects to adapt software from an old state to a new one that have been left incomplete. We value these migrations reaching an end state, whether it’s completing the planned migration or rolling back on an idea that won’t work. We discourage staying in an intermediary state.
+
+If you’re part of a team that has inherited an incomplete migration, we encourage you to consider likelihood of the migrations success and whether it fits into future goals and seek to bring it to a close.
+
+If you’re a team starting a migratory project, plan carefully. Minimise time spent in an intermediary state, by deferring anything non-essential to the migration, particularly if the migration is going to take more than a couple of weeks (most migrations take longer than expected). Be extremely careful starting a migration that may take longer than the foreseeable roadmap, you must be sure there will remain a business priority to fund it. Consider that the longer a migration is put off, the longer it’s likely to take - keep tasks like version upgrades small by doing them often.
+
+#### Prefer identifying and implementing missing features over toil tasks and data manipulation
+
+Treat the need for developers to intervene in the running of an application as an efficiency smell, whether through doing things for users, running manual tasks or writing code to manipulate data. 
+
+Use the evidence of these tasks to understand whether applications are missing features or have resolvable limitations and, where appropriate, implement changes to reduce the need for these tasks.
+
+Avoid letting developer interventions blur the boundary of the service offered to users. Consider each time that a developer does something manually for a user as GOV.UK creating an undocumented feature that carries the expense of a developer’s time each usage. For frequent requests, that are not suitable as user features, document the boundaries of the service developers will provide and a runbook of how to achieve them.
```

Comment:
Is it worth adding an example to the last sentence? We don't really do formal runbooks at GOV.UK, but I imagine there's a "common tasks" 2nd line page in the Dev Docs we could link to.

Reply:
That's a good shout, this might drift slightly into aspirational which wasn't the intention Al and I had. Let me see if I can find a good example of us documenting a dev manual task and update this.